server/useragent/version_info.go

```diff
@@ -0,0 +1,55 @@
+// Package useragent provides helper functions to pack
+// version information into a single User-Agent header.
+package useragent
+
+import (
+	"strings"
+)
+
+// VersionInfo is used to model UserAgent versions.
+type VersionInfo struct {
+	Name    string
+	Version string
+}
+
+func (vi *VersionInfo) isValid() bool {
+	const stopChars = " \t\r\n/"
+	name := vi.Name
+	vers := vi.Version
+	if len(name) == 0 || strings.ContainsAny(name, stopChars) {
```

Comment:
Do you need the name check, Doesn't strings.ContainersAny("", stopChars) work?

Reply:
It doesn't work when i checked

Reply:
```
package main

import (
	"fmt"
	"strings"
)

func main() {
	var x string
	fmt.Println(strings.ContainsAny("", "dan"))
	fmt.Println(strings.ContainsAny(x, "dan"))
}
```
```
$ go run /tmp/t.go
false
false
```
Works for me